test/unit_test/api/apps/services/test_provider_api_service_verify_discovery.py

```
@@ -0,0 +1,238 @@
+
+    assert ok is False
+    assert f"No models found for provider '{PROVIDER}'" in message
+    assert "failed" not in message
+
+
+@pytest.mark.asyncio
+async def test_credentials_in_the_base_url_never_reach_the_caller(monkeypatch, caplog):
+    """Users paste endpoints carrying secrets, as userinfo or as a query token.
+    Neither the response nor the application log may repeat them, and the client
+    echoes the URL it was handed back inside the exception text."""
+    secret_url = "https://svc:s3cr3t-token@models.internal:8443/v1?api-key=AKIA-SECRET#frag"
```

Ruff (0.16.1)

[error] Possible hardcoded password assigned to: "secret_url"

(S105)